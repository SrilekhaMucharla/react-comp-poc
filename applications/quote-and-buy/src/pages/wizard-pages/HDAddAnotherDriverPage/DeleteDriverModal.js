import React from 'react';
import PropTypes from 'prop-types';
import { AnalyticsHDModal as HDModal } from '../../../web-analytics';
import * as messages from './HDAddAnotherDriver.messages';

const DeleteDriverModal = ({
    onCancel, onConfirm, onClose, driverName, show, pageMetadata
}) => (
    <HDModal
        webAnalyticsView={{ ...pageMetadata, page_section: messages.removeDriver }}
        webAnalyticsEvent={{ event_action: messages.removeDriver }}
        id="remove-driver-modal"
        show={show}
        headerText={messages.areYouSure}
        cancelLabel={messages.goBack}
        confirmLabel={messages.yes}
        onCancel={onCancel}
        onConfirm={onConfirm}
        onClose={onClose}
    >
        <span>
    Are you sure you want to remove
            {driverName ? <b>{` ${driverName}`}</b> : ' this driver'}
    ?
        </span>
    </HDModal>
);

DeleteDriverModal.propTypes = {
    onCancel: PropTypes.func.isRequired,
    onConfirm: PropTypes.func.isRequired,
    onClose: PropTypes.func.isRequired,
    driverName: PropTypes.string,
    show: PropTypes.bool,
    pageMetadata: PropTypes.shape({
        page_name: PropTypes.string.isRequired,
        page_type: PropTypes.string.isRequired,
        sales_journey_type: PropTypes.string.isRequired
    }).isRequired
};

DeleteDriverModal.defaultProps = {
    driverName: '',
    show: false
};

export default DeleteDriverModal;
