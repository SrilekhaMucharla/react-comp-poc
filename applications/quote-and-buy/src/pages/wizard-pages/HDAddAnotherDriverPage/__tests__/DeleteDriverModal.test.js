import React from 'react';
import { shallow } from 'enzyme';
import DeleteDriverModal from '../DeleteDriverModal';

describe('DeleteDriverModal', () => {
    // given
    const onCancel = jest.fn();
    const onConfirm = jest.fn();
    const onClose = jest.fn();
    const driverName = 'driver name';
    const id = 'remove-driver-modal';

    const getWrapper = (props = {}) => shallow(
        <DeleteDriverModal
            onCancel={onCancel}
            onConfirm={onConfirm}
            onClose={onClose}
            driverName={driverName}
            show
            {...props} />
    );

    it('should render correctly and match the snapshot', () => {
        // given
        const wrapper = getWrapper();
        // then
        expect(wrapper).toMatchSnapshot();
    });

    it('should display HDModal', () => {
        // given
        const wrapper = getWrapper();
        // then
        expect(wrapper.find(`#${id}`)).toHaveLength(1);
    });

    it('should contain driver name', () => {
        // given
        const wrapper = getWrapper();
        // then
        expect(wrapper.find('span').text()).toContain(driverName);
    });

    it('should contain "this driver"', () => {
        // given
        const noDriverString = 'this driver';
        const wrapper = shallow(
            <DeleteDriverModal
                onCancel={onCancel}
                onConfirm={onConfirm}
                onClose={onClose}
                driverName=""
                show />
        );
        // then
        expect(wrapper.find('span').text()).toContain(noDriverString);
    });

    it('should pass functions to HDModal', () => {
        // given
        const wrapper = getWrapper();
        const modalProps = wrapper.find(`#${id}`).props();
        // then
        expect(modalProps.onCancel).toBe(onCancel);
        expect(modalProps.onClose).toBe(onClose);
        expect(modalProps.onConfirm).toBe(onConfirm);
    });
});
