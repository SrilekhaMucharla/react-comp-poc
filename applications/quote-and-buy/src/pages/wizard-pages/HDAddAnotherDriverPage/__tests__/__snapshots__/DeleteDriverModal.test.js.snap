// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteDriverModal should render correctly and match the snapshot 1`] = `
<ForwardRef
  cancelLabel="Go back"
  confirmLabel="Yes"
  headerText="Are you sure?"
  id="remove-driver-modal"
  onBeforeOpen={null}
  onCancel={[MockFunction]}
  onClose={[MockFunction]}
  onConfirm={[MockFunction]}
  show={true}
  webAnalyticsEvent={
    Object {
      "event_action": "Remove driver",
    }
  }
  webAnalyticsView={
    Object {
      "page_section": "Remove driver",
    }
  }
>
  <span>
    Are you sure you want to remove
    <b>
       driver name
    </b>
    ?
  </span>
</ForwardRef>
`;
